core/scripts/chaincli/handler/scrape_node_config.go

```diff
@@ -0,0 +1,270 @@
+	}
+	for i, ci := range node.CSAKeys {
+		if !ci.Equals(ni.CSAKeys[i]) {
+			diffs++
+			log.Errorf("CSA Info differs. The node returns %s but weiwatcher has %s", ci, ni.CSAKeys[i])
+		}
+	}
+
+	if !cmp.Equal(node.Ocr2ID, ni.Ocr2ID) {
+		diffs++
+		log.Errorf("OCR2 ID differs. The node returns %s but weiwatcher has %s", node.Ocr2ID, ni.Ocr2ID)
+	}
+
+	if !cmp.Equal(node.NodeAddress, ni.NodeAddress) {
+		diffs++
+		log.Errorf("Node address differs. The node returns %s but weiwatcher has %s", node.NodeAddress, ni.NodeAddress)
+	}
+
+	// preprocess the Peer ID from node bc it has p2p_ prefix
+	var peerIds []string
+	for _, pid := range node.PeerId {
+		peerIds = append(peerIds, pid[4:])
+	}
+	if !cmp.Equal(peerIds, ni.PeerId) {
+		diffs++
+		log.Errorf("Peer Id differs. The node returns %s but weiwatcher has %s", node.PeerId, ni.PeerId)
+	}
+
+	if !cmp.Equal(node.Ocr2OffchainPublicKey, ni.Ocr2OffchainPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Offchain Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2OffchainPublicKey, ni.Ocr2OffchainPublicKey)
+	}
+
+	if !cmp.Equal(node.Ocr2OnchainPublicKey, ni.Ocr2OnchainPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Onchain Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2OnchainPublicKey, ni.Ocr2OnchainPublicKey)
+	}
+
+	if !cmp.Equal(node.Ocr2ConfigPublicKey, ni.Ocr2ConfigPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Config Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2ConfigPublicKey, ni.Ocr2ConfigPublicKey)
+	}
+
+	return diffs == 0
+}
+
+func (h *baseHandler) ScrapeNodes() {
+	log, closeLggr := logger.NewLogger()
+	logger.Sugared(log).ErrorIfFn(closeLggr, "Failed to close logger")
+
+	ctx := context.Background()
+	h.scrapeNodes(ctx, log)
+}
+
+func (h *baseHandler) scrapeNodes(ctx context.Context, log logger.Logger) {
+	log.Warn("This scrapes node address, peer ID, CSA node address, CSA public key, OCR2 ID, OCR2 config pub key, OCR2 onchain pub key, and OCR2 offchain pub key.")
+	log.Warn("This does NOT scrape for payee address, admin address etc. Please verify that manually.")
+	cls := make([]cmd.HTTPClient, len(h.cfg.KeeperURLs))
+	for i := range h.cfg.KeeperURLs {
+		url := h.cfg.KeeperURLs[i]
+		email := h.cfg.KeeperEmails[i]
+		if len(email) == 0 {
+			email = defaultChainlinkNodeLogin
+		}
+		pwd := h.cfg.KeeperPasswords[i]
+		if len(pwd) == 0 {
+			pwd = defaultChainlinkNodePassword
+		}
+
+		cl, err := authenticate(url, email, pwd, log)
+		if err != nil {
+			log.Fatal(err)
+		}
+		cls[i] = cl
+	}
+
+	nodes := map[string]*NodeInfo{}
+	var wg sync.WaitGroup
+	for i, cl := range cls {
+		wg.Add(1)
+		go h.scrapeNodeInfo(&wg, i, cl, nodes, log)
+	}
+	wg.Wait()
+
+	// if node info is not in RDD and weiwatchers, don't proceed further
+	if !h.cfg.VerifyNodes {
+		return
+	}
+	nodeInfos := h.fetchNodeInfos(ctx, log)
+	cnt := 0
+	for _, ni := range nodeInfos {
+		nodeAddr := ni.NodeAddress[0]
```

Comment:
Should this be wrapped in a check that there are 1 or more elements in the `NodeAddress` slice and continue otherwise?

Reply:
Also is there a particular significance to the zeroth Node Address? If so I would maybe maybe a function on the node receiver to fetch this?